src/Compilers/Core/Portable/PEWriter/ManagedResource.cs

```diff
@@ -48,7 +48,12 @@ public void WriteData(BlobBuilder resourceWriter)
                         var count = (int)(stream.Length - stream.Position);
                         resourceWriter.WriteInt32(count);
 
-                        resourceWriter.WriteBytes(stream, count);
+                        int bytesWritten = resourceWriter.TryWriteBytes(stream, count);
+                        if (bytesWritten != count)
+                        {
+                            throw new EndOfStreamException(
```

Comment:
We need to have a test that goes through this line in order to verify how and where this exception is getting propagated.